### PR TITLE
Instant Search: Implement minimal search results and spelling correction

### PR DESCRIPTION
**Pull request by gibrown**
This is the same as #13313 but merging into instant-search-master rather than master.

## Changes proposed in this Pull Request:
Implementing the minimal search results mockup from p3QzjZ-VL-p2

There will be three different types of search results:
- minimal
- product
- engagement

But this is just focusing on the first one and setting us up for the other two.

This PR also implements the logic for corrected search queries when we do any spelling correction and displaying the total number of results. Here is what it all looks like:

Screen Shot 2019-08-26 at 8 24 06 PM

Instant Search remains gated by a `JETPACK_SEARCH_PROTOTYPE` define.

## Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Yes, this is a new feature being added onto Jetpack Instant Search (p3QzjZ-V3-p2).

## Testing instructions:
* Add `define( "JETPACK_SEARCH_PROTOTYPE", true );` to your wp-config.php.
* Ensure that your site has the Jetpack Pro plan and has Jetpack Search enabled.
* Enter a query into a search widget. Alternatively, navigate to a search page like `/?s=privacy`.
* Verify how the page looks and that you are getting results as expected 

You can also follow the instructions in #13290 to run your query against any site rather than the site you deploy the code on (all the data comes from the API).

## Proposed changelog entry for your changes:
* No changelog entry necessary; these changes will not be accessible to our users.

### CODE REVIEW

File: modules/search/instant-search/components/api.js [hunk 1]
```diff
@@ -3,26 +3,73 @@
...
-export function search( siteId, query ) {
-	return fetch( getAPIUrl( siteId, query ) );
+	const obj = {
+		query: query,
+		fields: fields,
+		highlight_fields: highlight_fields,
+		aggregations: aggs,
+		filter: filter,
+	};
+	return fetch(
```
Comment (bluefuton) [inline modules/search/instant-search/components/api.js]: How should we handle failed API requests?

Comment (gibrown) [reply modules/search/instant-search/components/api.js]: Good question. Probably depends on what the error code was. Let's deal with it in #13366

**Comment by bluefuton** [comment]
Layout bug with long titles:

Screen Shot 2019-08-30 at 13 50 57

**Review by bluefuton — APPROVED** [review]
Happy for us to merge this into the feature branch as a work-in-progress. Let's make sure we go back and address feedback from the original PR https://github.com/Automattic/jetpack/pull/13313 later.

**Comment by bluefuton** [comment]
Spelling suggestions seem to work well:

Screen Shot 2019-08-30 at 13 49 15

**Review by jeherve — APPROVED** [review]
LGTM 👍

**Pull request merged**